src/datasource.ts

```diff
@@ -343,17 +344,20 @@ export function filterEmptyQueryTargets(targets: QueryTarget[]): QueryTarget[] {
     }
   });
 
-  return targets.filter(target => {
-    if (target && !target.hide) {
-      if (target.tab === Timeseries || target.tab === undefined) {
-        return target.target;
+  return targets.reduce(
+    (result, target) => {
+      if (!target || target.hide) {
+        return result;
       }
-      if (target.tab === Asset || target.tab === Custom) {
-        return target.assetQuery && target.assetQuery.target;
-      }
-    }
-    return false;
-  });
+
+      const { tab, target: tsTarget, assetQuery } = target;
+
+      const value = tab === Asset || tab === Custom ? assetQuery && assetQuery.target : tsTarget;
+
+      return value && value !== '' ? ([...result, target] as QueryTarget[]) : result;
+    },
+    [] as QueryTarget[]
```

Comment:
why reduce? I tried to get rid of this cumbersome `reduce()` with `[...result, value]`
and now you introducing them again :)
why we cannot keep filter?

Reply:
I've been doing manipulations with targets during the development and then returned to simple filtering, just didn't move back to filter

Reply:
can we move back? there is always map for manipulations, more readable

Reply:
yes, but when you do manipulation and filtering its better to implement via reduce to drop complexity. I'm reverting this